src/main/java/io/github/syst3ms/skriptparser/expressions/CondExprEmpty.java

```diff
@@ -0,0 +1,64 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import java.util.Arrays;
+
+import org.jetbrains.annotations.Nullable;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.base.ConditionalExpression;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+
+/**
+ * Check if a given string or list is empty.
+ *
+ * @name Empty
+ * @pattern %strings% (is|are)[1:( not|n't)] empty
+ * @pattern %objects% is[1:( not|n't)] [a[n]] empty list
+ * @since ALPHA
+ * @author Olyno
+ */
+public class CondExprEmpty extends ConditionalExpression {
+
+    static {
+        Parser.getMainRegistration().addExpression(CondExprEmpty.class,
+            Boolean.class,
+            true,
+            2,
+            "%strings% (is|are)[1:( not|n't)] empty",
+            "%objects% is[1:( not|n't)] [a[n]] empty list"
+        );
+    }
+
+    private Expression<?> expr;
+    private boolean comparingList;
+
+    @Override
+    public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+        expr = (Expression<?>) expressions[0];
+        comparingList = matchedPattern == 1;
+        setNegated(parseContext.getParseMark() == 1);
+        return true;
+    }
+
+    @Override
+    public boolean check(TriggerContext ctx) {
+        Object[] values = expr.getValues(ctx);
+        if (values.length == 0)
+            return !isNegated();
+        if (values.length == 1) {
```

Comment:
I think this can all be removed. Just check it in a for loop or use Stream#allMatch(). There is no need to do a separate check for one value as the check is the same as for multiple values.